codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -46,24 +50,20 @@
  */
 public class DefaultHeaders<K, V, T extends Headers<K, V, T>> implements Headers<K, V, T> {
     /**
-     * How big the underlying array is for the hash data structure.
-     * <p>
-     * This should be a power of 2 so the {@link #index(int)} method can full address the memory.
+     * Enforce an upper bound of 128 because {@link #hashMask} is a byte.
+     * The max possible value of {@link #hashMask} is one less than this value.
      */
-    private static final int ARRAY_SIZE = 1 << 4;
-    private static final int HASH_MASK = ARRAY_SIZE - 1;
-    static final int HASH_CODE_SEED = 0xc2b2ae35; // constant borrowed from murmur3
+    private static final int ENTRIES_MAX_SIZE = min(128,
```

Comment:
@nmittler - The existing name is confusing. I will change the name to `ARRAY_SIZE_HINT_MAX` to clarify ... does this help?